src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -1109,6 +1126,58 @@ private void AddPatchNode(Patch patch, Vector2 position)
 
         patch.ApplyPatchEventVisuals(node);
 
+        var playerSpecies = patch.FindSpeciesByID(PlayerSpeciesID);
+        if (playerSpecies != null)
+        {
+            var playerPopulationIndicatorAmount = (int)Math.Ceiling(
+                patch.GetSpeciesSimulationPopulation(playerSpecies) * 0.004);
+
+            var indicatorExcess = Math.Clamp(playerSpeciesPopulationIndicators.Count - playerPopulationIndicatorAmount,
+                0,
+                playerSpeciesPopulationIndicators.Count);
+
+            for (int i = 0; i < indicatorExcess; ++i)
+            {
+                playerSpeciesPopulationIndicators.Last().QueueFree();
+            }
+
+            // Trip the list to keep it clean of disposed objects
+            var range = playerSpeciesPopulationIndicators.Count - indicatorExcess;
+            if (range > 0)
+                playerSpeciesPopulationIndicators = playerSpeciesPopulationIndicators.GetRange(0, range);
+
+            for (int i = 0; i < playerPopulationIndicatorAmount; ++i)
+            {
+                var noCached = i >= playerSpeciesPopulationIndicators.Count;
+
+                Control indicator;
+                if (noCached)
+                {
+                    indicator = populationIndicatorScene.Instantiate<Control>();
+                }
+                else
+                {
+                    indicator = playerSpeciesPopulationIndicators[i];
+                }
+
+                indicator.MouseFilter = MouseFilterEnum.Ignore;
+
+                var nodeModifier = node.Position.LengthSquared();
+                var modifierSinus = MathF.Sin(i);
+
+                if (noCached)
+                {
+                    populationIndicatorContainer.AddChild(indicator);
```

Comment:
This could be combined with the above if where this is instantiated for easier reading of this block of code.